pylint/checkers/variables.py

```diff
@@ -2139,388 +1749,47 @@ def _ignored_modules(self) -> Iterable[str]:
-        return (VariableVisitConsumerAction.RETURN, found_nodes)
-
-    @staticmethod
-    def _is_never_evaluated(
-        defnode: nodes.NamedExpr,
-        defnode_parent: nodes.IfExp,
-    ) -> bool:
-        """Check if a NamedExpr is inside a side of if ... else that never
-        gets evaluated.
-        """
-        inferred_test = utils.safe_infer(defnode_parent.test)
-        if isinstance(inferred_test, nodes.Const):
-            if inferred_test.value is True and defnode == defnode_parent.orelse:
-                return True
-            if inferred_test.value is False and defnode == defnode_parent.body:
-                return True
-        return False
-
-    @staticmethod
-    def _is_variable_annotation_in_function(node: nodes.NodeNG) -> bool:
-        is_annotation = utils.get_node_first_ancestor_of_type(node, nodes.AnnAssign)
-        return (
-            is_annotation
-            and utils.get_node_first_ancestor_of_type(  # type: ignore[return-value]
-                is_annotation, nodes.FunctionDef
-            )
-        )
-
-    def _ignore_class_scope(self, node: nodes.NodeNG) -> bool:
-        """Return True if the node is in a local class scope, as an assignment.
-
-        Detect if we are in a local class scope, as an assignment.
-        For example, the following is fair game.
-
-        class A:
-           b = 1
-           c = lambda b=b: b * b
-
-        class B:
-           tp = 1
-           def func(self, arg: tp):
-               ...
-        class C:
-           tp = 2
-           def func(self, arg=tp):
-               ...
-        class C:
-           class Tp:
-               pass
-           class D(Tp):
-               ...
+        class B:
+           tp = 1
+           def func(self, arg: tp):
+               ...
+        class C:
+           tp = 2
+           def func(self, arg=tp):
+               ...
+        class C:
+           class Tp:
+               pass
+           class D(Tp):
+               ...
         """
         name = node.name
         frame = node.statement().scope()
-        in_annotation_or_default_or_decorator = self._defined_in_function_definition(
+        in_annotation_or_default_or_decorator = _defined_in_function_definition(
             node, frame
         )
         in_ancestor_list = utils.is_ancestor_name(frame, node)
-        if in_annotation_or_default_or_decorator or in_ancestor_list:
-            frame_locals = frame.parent.scope().locals
-        else:
-            frame_locals = frame.locals
+        frame_locals = (
+            frame.parent.scope().locals
+            if in_annotation_or_default_or_decorator or in_ancestor_list
+            else frame.locals
+        )
         return not (
             (isinstance(frame, nodes.ClassDef) or in_annotation_or_default_or_decorator)
-            and not self._in_lambda_or_comprehension_body(node, frame)
+            and not _in_lambda_or_comprehension_body(node, frame)
```

Comment:
This kind of change is okay with me: the VariablesChecker and NamesConsumer are long enough that having all these methods in a closure makes it harder to see what they do. But let's get a +1 from someone else.